utils/utils.go

```diff
@@ -0,0 +1,72 @@
+package utils
+
+import (
+	"net/url"
+	"strconv"
+
+	"github.com/ONSdigital/dp-code-list-api/models"
+)
+
+// GetPositiveIntQueryParameter obtains the positive int value of query var defined by the provided varKey
+func GetPositiveIntQueryParameter(queryVars url.Values, varKey string, defaultValue int) (val int, err error) {
+	strVal, found := queryVars[varKey]
+	if !found {
+		return defaultValue, nil
+	}
+	val, err = strconv.Atoi(strVal[0])
+	if err != nil {
+		return -1, err
+	}
+	if val < 0 {
+		return 0, nil
+	}
+	return val, nil
+}
+
+func CodelistsSlice(full []models.CodeList, offset, limit int) (sliced []models.CodeList) {
```

Comment:
These slice functions for each model type appear to be used only once. Consider moving these functions into the API package, in the file next to where they are used.